gk/main.c

```diff
@@ -745,18 +745,20 @@ gk_hash_add_flow_entry(struct gk_instance *instance,
 	struct ip_flow *flow, unsigned int request_timeout_cycles,
 	uint32_t rss_hash_val, enum gk_flow_state state_to_add)
 {
+	int retried = false;
 	while (true) {
 		int ret = rte_hash_add_key_with_hash(
 			instance->ip_flow_hash_table, flow, rss_hash_val);
 		if (ret == -ENOSPC) {
-			GK_LOG(WARNING,
-				"The GK block failed to add new key to hash table in %s due to lack of space\n",
-				__func__);
+			if (state_to_add == GK_REQUEST || retried)
```

Comment:
It's my understanding that if `drop_flow_entry_heuristically()` returns without an error, `rte_hash_add_key_with_hash()` should work on the following call. That is why there wasn't a variable to take the role of variable `retried`. But I see the value of variable `retried` as a bug catcher, so let's drop it from this `if` statement and add `RTE_VERIFY(!retried);` immediately before this `if` statement.